test/python/primitive_test.py

```diff
@@ -249,3 +249,9 @@ def test_service():
         PingService().ping(nonce=u'nonce')
     with raises(TypeError):
         PingService().ping(wrongkwd=u'a')
+
+
+def test_optional_initializer_test():
+    product = Product(name=u'coffee', sale=False)
+    assert product.price is None
+    assert product.url is None
```

Comment:
We should test union tags as well e.g.:

~~~~ python
def test_union_tags_optional_initializer():
    t = TagB(a=u'test', c=123)
    assert t.a == u'test'
    assert t.b is None
    assert t.c == 123
    assert t.d is None
~~~~